compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -3377,7 +3383,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
       else if (ctx.mode.is(Mode.Pattern))
         typedUnApply(cpy.Apply(tree)(op, l :: r :: Nil), pt)
       else {
-        val app = typedApply(desugar.binop(l, op, r), pt)
+        val app = typedApply(desugar.binop(l, op, r).withAttachmentsFrom(tree), pt)
```

Comment:
Needed to add `withAttachmentsFrom(tree)` here so that we don't throw away attachments when desugaring binary operations.